integration/dhclient_test.go

```diff
@@ -76,42 +128,77 @@ func TestDhclient(t *testing.T) {
 	defer ccleanup()
 
 	if err := dhcpClient.Expect("Configured eth0 with IPv4 DHCP Lease"); err != nil {
+		t.Errorf("%s: %v", testutil.NowLog(), err)
+	}
+	if err := dhcpClient.Expect("inet 192.168.0.10"); err != nil {
+		t.Errorf("%s: %v", testutil.NowLog(), err)
+	}
+}
+
+func TestQEMUDHCPTimesOut(t *testing.T) {
+	// TODO: support arm
+	if vmtest.TestArch() != "amd64" {
+		t.Skipf("test not supported on %s", vmtest.TestArch())
+	}
+
+	network := qemu.NewNetwork()
+	dhcpClient, ccleanup := vmtest.QEMUTest(t, &vmtest.Options{
+		Name: "TestQEMUDHCPClient",
+		BuildOpts: uroot.Opts{
+			Commands: uroot.BusyBoxCmds(
+				"github.com/u-root/u-root/cmds/core/ip",
+				"github.com/u-root/u-root/cmds/core/init",
+				"github.com/u-root/u-root/cmds/core/echo",
+				"github.com/u-root/u-root/cmds/core/dhclient",
+				"github.com/u-root/u-root/cmds/core/shutdown",
+			),
+		},
+		QEMUOpts: qemu.Options{
+			SerialOutput: vmtest.TestLineWriter(t, "client"),
+			Timeout:      50 * time.Second,
+			Devices: []qemu.Device{
+				network.NewVM(),
+			},
+		},
+		Uinit: []string{
+			"dhclient -v -retry 2 -timeout 10",
+			"echo \"DHCP timed out\"",
+			"shutdown -h",
+		},
+	})
+	defer ccleanup()
+
+	if err := dhcpClient.Expect("Could not configure eth0 for IPv4: no matching response packet received"); err != nil {
 		t.Error(err)
 	}
-	if err := dhcpClient.Expect("inet 192.168.0.2"); err != nil {
+	if err := dhcpClient.Expect("Could not configure eth0 for IPv6: no matching response packet received"); err != nil {
+		t.Error(err)
+	}
+	if err := dhcpClient.Expect("DHCP timed out"); err != nil {
 		t.Error(err)
 	}
 }
 
-// TestPxeboot runs a server and client to test pxebooting a node.
-// TODO: FIX THIS TEST!
-// Change the t.Logf below back to t.Errorf
-func TestPxeboot(t *testing.T) {
+func TestDHCP6(t *testing.T) {
```

Comment:
Is it implied that dhcp4 is basically tested by testpxeboot which is why we don't need a separate test for it? I would like to have a separate one if possible

Reply:
it's back.